### PR TITLE
51 search implementation

### LINKED ISSUE #51: [Search] Implement search
- Show first 4 products in product block
- Show first 4 recipes in recipe block
- Render remaining search results in animation view
- Localize the labels using placeholder
- Fix the See all button
- Remove see all button from "Other" block in mobile view
- Show all items in other block in mobile view, add scroll bar.

Reference code: https://github.com/adobe/helix-website/blob/main/blocks/header/gnav-search.js#L58

### PR DESCRIPTION
**Pull request by noormd08**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #51 

Test URLs:
- Before: https://main--fonterra-fernleaf--hlxsites.hlx.page/
- After: https://51-search-implementation--fonterra-fernleaf--hlxsites.hlx.page/en

### CODE REVIEW

File: blocks/header/search.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import { fetchSearch } from '../../scripts/scripts.js';
+import { fetchPlaceholders } from '../../scripts/lib-franklin.js';
+
+/**
+ * Clear the category wise search results
+ */
+export function clearSearchResults() {
+  const searchProducts = ['product', 'recipe', 'story'];
+  const searchContainer = document.querySelector('#search-dialog .search-results-container');
```
Comment (sselvara) [inline blocks/header/search.js:9]: Simplify this pls.
 
 ```
searchProducts.forEach((category) => {
    document.querySelector(`#search-dialog .search-results-container .${category}-category-list`).innerHTML = ""
 });
```

File: blocks/header/search.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    const placeholders = await fetchPlaceholders();
+    const searchData = await fetchSearch();
+    spinner.style.display = 'none';
+    if (placeholders && searchData) {
+      let filteredResults = [];
+      productCategories.map((category) => {
+        filteredResults = searchData.filter((el) =>
+          /* eslint implicit-arrow-linebreak: ["error", "below"] */
+          /* eslint max-len: ["error", { "code": 300 }] */
+          el.category.toLowerCase() === category.toLowerCase() && (el.title.toLowerCase().includes(searchValue) || el.shorttitle.toLowerCase().includes(searchValue) || el.description.toLowerCase().includes(searchValue) || el.tags.toLowerCase().includes(searchValue)));
```
Comment (sselvara) [inline blocks/header/search.js:100]: consider using array to push the values and check include once.

File: blocks/header/search.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    const spinner = document.querySelector('#search-dialog .overlay-loading');
+    spinner.style.display = 'block';
+    const placeholders = await fetchPlaceholders();
+    const searchData = await fetchSearch();
+    spinner.style.display = 'none';
+    if (placeholders && searchData) {
+      let filteredResults = [];
+      productCategories.map((category) => {
+        filteredResults = searchData.filter((el) =>
+          /* eslint implicit-arrow-linebreak: ["error", "below"] */
```
Comment (sselvara) [inline blocks/header/search.js:98]: use `{}` to avoid this error instead of disabling it.

**Review by sselvara — APPROVED** [review]

File: blocks/header/search.js (hunk 1 above)
Comment (noormd08) [reply blocks/header/search.js:9]: Agree. Somehow missed it. Will consider as a part of separate ticket

File: blocks/header/search.js (hunk 2 above)
Comment (noormd08) [reply blocks/header/search.js:100]: Will consider as a part of separate ticket

File: blocks/header/search.js (hunk 3 above)
Comment (noormd08) [reply blocks/header/search.js:98]: Will consider as a part of separate ticket

**Pull request merged**